File: api/src/common/entity/show.ts
import { Column, CreateDateColumn, Entity, OneToMany, PrimaryGeneratedColumn } from 'typeorm';

import { Season } from './season';

@Entity()
export class Show {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ unique: true })
  providerId: string;

  @Column()
  name: string;

  @Column({ nullable: true })
  providerUrl?: string;

  @Column({ nullable: true, length: 4096 })
  summary?: string;

  @Column({ nullable: true })
  language?: string;

  @Column({ nullable: true })
  status?: string;

  @Column({ nullable: true })
  runtime?: number;

  @Column({ nullable: true })
  premiered?: Date;

  @Column({ nullable: true })
  ended?: Date;

  @Column({ nullable: true })
  officialSite?: string;

  @Column({ nullable: true })
  network?: string;

  @Column({ nullable: true })
  imageMedium?: string;

  @Column({ nullable: true })
  imageOriginal?: string;

  @Column({ nullable: true })
  imdbId?: string;

  @OneToMany(() => Season, season => season.show, {
    cascade: true,
  })
  seasons?: Season[];

  @CreateDateColumn()
  createdAt?: Date;
}

File: api/models/db/show.go
package modelsdb

import (
	"time"
)

type Show struct {
	Model
	ProviderID    *string    `json:"providerId"`
	ProviderUrl   *string    `json:"providerUrl"`
	Name          string     `json:"name" gorm:"not null"`
	Summary       *string    `json:"summary" gorm:"type:text"`
	Language      *string    `json:"language"`
	Status        *string    `json:"status"`
	Runtime       *uint16    `json:"runtime"`
	Premiered     *time.Time `json:"premiered"`
	Ended         *time.Time `json:"ended"`
	OfficialSite  *string    `json:"officialSite"`
	Network       *string    `json:"network"`
	ImageMedium   *string    `json:"imageMedium"`
	ImageOriginal *string    `json:"imageOriginal"`
	ImdbId        *string    `json:"imdbId"`
	Seasons       []Season   `json:"seasons"`
	Completed     *int       `gorm:"-" json:"completed"`
}
